<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>seqArchR: A package for de novo discovery of different sequence
architectures — seqArchR • seqArchR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="seqArchR: A package for de novo discovery of different sequence
architectures — seqArchR"><meta property="og:description" content="Given a set of DNA sequences, seqArchR enables unsupervised
discovery of _de novo_ clusters with characteristic sequence
architectures characterized by position-specific motifs or composition
of stretches of nucleotides, e.g., CG-richness, etc.
Call this function to process a data set using seqArchR."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-67848084-3"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-67848084-3');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">seqArchR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/seqArchR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/snikumbh/seqArchR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>seqArchR: A package for de novo discovery of different sequence
architectures</h1>
    <small class="dont-index">Source: <a href="https://github.com/snikumbh/seqArchR/blob/HEAD/R/seqArchR.R" class="external-link"><code>R/seqArchR.R</code></a>, <a href="https://github.com/snikumbh/seqArchR/blob/HEAD/R/seqArchR_main.R" class="external-link"><code>R/seqArchR_main.R</code></a></small>
    <div class="hidden name"><code>seqArchR.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given a set of DNA sequences, <code>seqArchR</code> enables unsupervised
discovery of _de novo_ clusters with characteristic sequence
architectures characterized by position-specific motifs or composition
of stretches of nucleotides, e.g., CG-richness, etc.</p>
<p>Call this function to process a data set using seqArchR.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">seqArchR</span><span class="op">(</span>
  <span class="va">config</span>,
  <span class="va">seqs_ohe_mat</span>,
  <span class="va">seqs_raw</span>,
  seqs_pos <span class="op">=</span> <span class="cn">NULL</span>,
  total_itr <span class="op">=</span> <span class="cn">NULL</span>,
  set_ocollation <span class="op">=</span> <span class="cn">NULL</span>,
  fresh <span class="op">=</span> <span class="cn">TRUE</span>,
  use_oc <span class="op">=</span> <span class="cn">NULL</span>,
  o_dir <span class="op">=</span> <span class="cn">NULL</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>config</dt>
<dd><p>seqArchR configuration object as returned by
<code><a href="set_config.html">set_config</a></code>. This is a required argument.</p></dd>
<dt>seqs_ohe_mat</dt>
<dd><p>A matrix of one-hot encoded sequences with sequences
along columns. This is a required argument.</p></dd>
<dt>seqs_raw</dt>
<dd><p>A <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></code> object. The FASTA
sequences as a DNAStringSet object. This argument required argument.</p></dd>
<dt>seqs_pos</dt>
<dd><p>Vector. Specify the tick labels for sequence positions.
Default is NULL.</p></dd>
<dt>total_itr</dt>
<dd><p>Numeric. Specify the number of iterations to perform. This
should be greater than zero. Default is NULL.</p></dd>
<dt>set_ocollation</dt>
<dd><p>Logical vector. A logical vector of length `total_itr`
specifying for every iteration of seqArchR if collation of clusters from
outer chunks should be performed. TRUE denotes clusters are collated,
FALSE otherwise.</p></dd>
<dt>fresh</dt>
<dd><p>Logical. Specify if this is (not) a fresh run. Because
seqArchR enables checkpointing, it is possible to perform additional
iterations upon clusters from an existing seqArchR result (or a checkpoint)
object. See 'use_oc' argument.
For example, when processing a set of FASTA sequences,
if an earlier call to seqArchR performed two iterations, and now you wish to
perform a third, the arguments `fresh` and `use_oc` can be used. Simply set
`fresh` to FALSE and assign the sequence clusters from iteration two from
the earlier result to `use_oc`. As of v0.1.3, with this setting, seqArchR
returns a new result object as if the additional iteration performed is the
only iteration.</p></dd>
<dt>use_oc</dt>
<dd><p>List. Clusters to be further processed with seqArchR. These can
be from a previous seqArchR result (in which case use
<code><a href="get_seqs_clust_list.html">get_seqs_clust_list</a></code> function), or simply clusters from any
other method.
Warning: This has not been rigorously tested yet (v0.1.3).</p></dd>
<dt>o_dir</dt>
<dd><p>Character. Specify the output directory with its path. seqArchR
will create this directory. If a directory with the given name exists at the
given location, seqArchR will add a suffix to the directory name. This
change is reported to the user. Default is NULL. When NULL, just the result
is returned, and no plots or checkpoints or result is written to disk.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A nested list of elements as follows:</p><dl><dt>seqsClustLabels</dt>
<dd><p>A list with cluster labels for all sequences per
iteration of seqArchR. The cluster labels as stored as characters.</p></dd>


<dt>clustBasisVectors</dt>
<dd><p>A list with information on NMF basis vectors per
iteration of seqArchR. Per iteration, there are two variables `nBasisVectors`
storing the number of basis vectors after model selection,
and `basisVectors`, a matrix storing the basis vectors themselves.
Dimensions of the `basisVectors` matrix are 4*L x nBasisVectors
(mononucleotide case) or 16*L x nBasisVectors (dinucleotide case).</p></dd>


<dt>clustSol</dt>
<dd><p>The clustering solution obtained upon processing the raw
clusters from the last iteration of seqArchR's result. This is handled
internally by the function <code><a href="collate_seqArchR_result.html">collate_seqArchR_result</a></code> using the
default setting of Euclidean distance and ward.D linkage hierarchical
clustering.</p></dd>


<dt>rawSeqs</dt>
<dd><p>The input sequences as a DNAStringSet object.</p></dd>


<dt>timeInfo</dt>
<dd><p>Stores the time taken (in minutes) for processing each
iteration. This element is added only if `time` flag is set to TRUE in
config.</p></dd>


<dt>config</dt>
<dd><p>The configuration used for processing.</p></dd>

<dt>call</dt>
<dd><p>The function call itself.</p></dd>


</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>The seqArchR package provides three categories of important functions:
related to data preparation and manipulation, performing non-negative
matrix factorization, performing clustering, and visualization-related
functions.</p>
    </div>
    <div id="functions-for-data-preparation-and-manipulation">
    <h2>Functions for data preparation and manipulation</h2>
    

<ul><li><p><code><a href="prepare_data_from_FASTA.html">prepare_data_from_FASTA</a></code></p></li>
<li><p><code><a href="get_one_hot_encoded_seqs.html">get_one_hot_encoded_seqs</a></code></p></li>
</ul></div>
    <div id="functions-for-visualizations">
    <h2>Functions for visualizations</h2>
    

<ul><li><p><code><a href="plot_arch_for_clusters.html">plot_arch_for_clusters</a></code></p></li>
<li><p><code><a href="plot_ggseqlogo_of_seqs.html">plot_ggseqlogo_of_seqs</a></code></p></li>
<li><p><code><a href="viz_bas_vec_heatmap_seqlogo.html">viz_bas_vec_heatmap</a></code></p></li>
<li><p><code><a href="viz_bas_vec_heatmap_seqlogo.html">viz_bas_vec_seqlogo</a></code></p></li>
<li><p><code><a href="viz_bas_vec_heatmap_seqlogo.html">viz_bas_vec_heatmap_seqlogo</a></code></p></li>
<li><p><code><a href="viz_seqs_acgt_mat.html">viz_seqs_acgt_mat</a></code></p></li>
<li><p><code><a href="plot_ggseqlogo.html">plot_ggseqlogo</a></code></p></li>
<li><p><code><a href="plot_ggheatmap.html">plot_ggheatmap</a></code></p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="va">fname</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"example_data.fa.gz"</span>,</span>
<span class="r-in">                        package <span class="op">=</span> <span class="st">"seqArchR"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Specifying 'dinuc' generates dinucleotide features</span></span>
<span class="r-in"><span class="va">inputSeqsMat</span> <span class="op">&lt;-</span> <span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="prepare_data_from_FASTA.html">prepare_data_from_FASTA</a></span><span class="op">(</span>fasta_fname <span class="op">=</span> <span class="va">fname</span>,</span>
<span class="r-in">    sinuc_or_dinuc <span class="op">=</span> <span class="st">"dinuc"</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sequences OK, </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read 200 sequences</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Generating dinucleotide profiles</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">inputSeqsRaw</span> <span class="op">&lt;-</span> <span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="prepare_data_from_FASTA.html">prepare_data_from_FASTA</a></span><span class="op">(</span>fasta_fname <span class="op">=</span> <span class="va">fname</span>,</span>
<span class="r-in">    raw_seq <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Set seqArchR configuration</span></span>
<span class="r-in"><span class="va">seqArchRconfig</span> <span class="op">&lt;-</span> <span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="set_config.html">set_config</a></span><span class="op">(</span></span>
<span class="r-in">    parallelize <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span class="r-in">    n_cores <span class="op">=</span> <span class="fl">2</span>,</span>
<span class="r-in">    n_runs <span class="op">=</span> <span class="fl">100</span>,</span>
<span class="r-in">    k_min <span class="op">=</span> <span class="fl">1</span>,</span>
<span class="r-in">    k_max <span class="op">=</span> <span class="fl">20</span>,</span>
<span class="r-in">    mod_sel_type <span class="op">=</span> <span class="st">"stability"</span>,</span>
<span class="r-in">    bound <span class="op">=</span> <span class="fl">10</span><span class="op">^</span><span class="op">-</span><span class="fl">8</span>,</span>
<span class="r-in">    chunk_size <span class="op">=</span> <span class="fl">100</span>,</span>
<span class="r-in">    flags <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>debug <span class="op">=</span> <span class="cn">FALSE</span>, time <span class="op">=</span> <span class="cn">TRUE</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span class="r-in">        plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Run seqArchR</span></span>
<span class="r-in"><span class="va">seqArchRresult</span> <span class="op">&lt;-</span> <span class="fu">seqArchR</span><span class="fu">::</span><span class="fu">seqArchR</span><span class="op">(</span>config <span class="op">=</span> <span class="va">seqArchRconfig</span>,</span>
<span class="r-in">                          seqs_ohe_mat <span class="op">=</span> <span class="va">inputSeqsMat</span>,</span>
<span class="r-in">                          seqs_raw <span class="op">=</span> <span class="va">inputSeqsRaw</span>,</span>
<span class="r-in">                          seqs_pos <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">100</span>,by<span class="op">=</span><span class="fl">1</span><span class="op">)</span>,</span>
<span class="r-in">                          total_itr <span class="op">=</span> <span class="fl">2</span>,</span>
<span class="r-in">                          set_ocollation <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Setting up ──────────────────────────────────────────────────────────────────</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Parallelization: 2 cores</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Model selection by factor stability</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Bound: 1e-08</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Iteration 1 of 2 [1 chunk]</span> <span style="color: #00BBBB;">──────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="font-weight: bold;">Outer chunk 1 of 1 [Size: 200]</span> ──</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Inner chunk 1 of 2 [Size: 100] </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking K =2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking K =3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking K =4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Best K for this chunk: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Adjusting for overfitting, fetched 2 clusters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Inner chunk 2 of 2 [Size: 100] </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking K =2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking K =3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking K =4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Best K for this chunk: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> 1 of 1 outer chunk complete</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> 1 of 2 iterations complete</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Iteration 1 completed: 1m 21.8s</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Time ellapsed since start: 1m 22s</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Iteration 2 of 2 [3 chunks]</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="font-weight: bold;">Outer chunk 1 of 3 [Size: 50]</span> ──</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Inner chunk 1 of 1 [Size: 50] </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking K =2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Best K for this chunk: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> 1 of 3 outer chunks complete</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="font-weight: bold;">Outer chunk 2 of 3 [Size: 55]</span> ──</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Inner chunk 1 of 1 [Size: 55] </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking K =2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking K =3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Best K for this chunk: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> 2 of 3 outer chunks complete</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="font-weight: bold;">Outer chunk 3 of 3 [Size: 95]</span> ──</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Inner chunk 1 of 1 [Size: 95] </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking K =2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking K =3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Best K for this chunk: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> 3 of 3 outer chunks complete</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> 2 of 2 iterations complete</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Iteration 2 completed: 52.7s</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Time ellapsed since start: 2m 14.7s</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── seqArchR exiting</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2m 14.8s ───────────────────────────────────────────────────</span>
<span class="r-in"></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Sarvesh Nikumbh.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

