<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Example usage of _seqArchR_ on simulated DNA sequences • seqArchR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Example usage of _seqArchR_ on simulated DNA sequences">
<meta property="og:description" content="seqArchR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-67848084-3"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-67848084-3');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">seqArchR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/seqArchR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/snikumbh/seqArchR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="seqArchR_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Example usage of <em>seqArchR</em> on simulated DNA sequences</h1>
                        <h4 class="author">Sarvesh Nikumbh</h4>
            
            <h4 class="date">2021-12-02</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/snikumbh/seqArchR/blob/master/vignettes/seqArchR.Rmd"><code>vignettes/seqArchR.Rmd</code></a></small>
      <div class="hidden name"><code>seqArchR.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><em>seqArchR</em> is a non-negative matrix factorization (NMF)-based unsupervised learning approach for identifying different core promoter sequence architectures. <em>seqArchR</em> implements an algorithm based on chunking and iterative processing. While matrix factorization-based applications are known to scale poorly for large amounts of data, seqArchR’s algorithm enables scalable processing of large number of sequences. A notable advantage of seqArchR is that the sequence motifs – the lengths and positional specificities of individual motifs, and complex inter-relationships where multiple motifs are at play in tandem, all are simultaneously inferred from the data. To our knowledge, this is a novel application of NMF on biological sequence data capable of simultaneously discovering the sequence motifs and their positions. For a more detailed discussion, see preprint/publication [TODO: link here].</p>
<p>This vignette demonstrates <em>seqArchR</em>’s usage with the help of a synthetic DNA sequences data set. Please refer to the paper (TODO: cite paper/preprint) for a detailed description of <em>seqArchR</em>’s algorithm. The paper also discusses the various parameters and their settings. For completeness, the following section gives a brief overview of the algorithm.</p>
</div>
<div id="seqarchrs-algorithm" class="section level1">
<h1 class="hasAnchor">
<a href="#seqarchrs-algorithm" class="anchor"></a><em>seqArchR</em>’s algorithm</h1>
<p><em>seqArchR</em> implements a chunking-based iterative procedure. Below is a schematic of seqArchR’s algorithm.</p>
<p><img src="seqArchR_algorithm_static_illustrator_cropped.png" width="550" align="center"></p>
<!-- ![Schematic of seqArchR's algorithm]() -->
<!-- ![Schematic of seqArchR's algorithm](seqArchR_algorithm_1080p_cropped.gif) -->
<p>Further details to follow.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p><em>seqArchR</em> is currently made available via GitHub, thus, you can use the following procedure for installing <em>seqArchR</em>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"remotes"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span>   
<span class="op">}</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"snikumbh/seqArchR"</span><span class="op">)</span>
 </code></pre></div>
<p>In case of any errors, please consider looking up: <a href="https://github.com/snikumbh/seqArchR">https://github.com/snikumbh/seqArchR</a>. If none of the already noted points with regards to troubleshooting <em>seqArchR</em>’s installation help, please file a <a href="https://github.com/snikumbh/seqArchR/issues/new">new issue</a>.</p>
</div>
<div id="working-with-seqarchr" class="section level1">
<h1 class="hasAnchor">
<a href="#working-with-seqarchr" class="anchor"></a>Working with <em>seqArchR</em>
</h1>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load seqArchR</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://snikumbh.github.io/seqArchR/">seqArchR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">#&gt;     union, unique, unsplit, which.max, which.min</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid, I, unname</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'Biostrings'</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     strsplit</span>


<span class="co"># Set seed for reproducibility</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></code></pre></div>
<div id="synthetic-data-explained" class="section level2">
<h2 class="hasAnchor">
<a href="#synthetic-data-explained" class="anchor"></a>Synthetic data explained</h2>
<p>In order to demonstrate the efficacy of <em>seqArchR</em>, we use <em>seqArchR</em> to cluster DNA sequences in a synthetic data set which was generated as follows. A set of 200 simulated DNA sequences was generated, each 100 nucleotides long and with uniform probability for all nucleotides. These sequences have four clusters in them, each with 50 sequences. The profiles of the four clusters are:</p>
<table class="table">
<thead><tr class="header">
<th>Cluster</th>
<th align="left">Characteristic Motifs</th>
<th align="left">Motif Occurrence Position</th>
<th>#Sequences</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>A</td>
<td align="left">Dinucleotide repeat <code>AT</code>
</td>
<td align="left">every 10 nt</td>
<td>50</td>
</tr>
<tr class="even">
<td>B</td>
<td align="left"><code>GATTACA</code></td>
<td align="left">40</td>
<td>50</td>
</tr>
<tr class="odd">
<td></td>
<td align="left"><code>GAGAG</code></td>
<td align="left">60</td>
<td></td>
</tr>
<tr class="even">
<td>C</td>
<td align="left"><code>GAGAG</code></td>
<td align="left">60</td>
<td>50</td>
</tr>
<tr class="odd">
<td>D</td>
<td align="left"><code>GAGAG</code></td>
<td align="left">80</td>
<td>50</td>
</tr>
<tr class="even">
<td></td>
<td align="left"><code>TCAT</code></td>
<td align="left">40</td>
<td></td>
</tr>
</tbody>
</table>
<!-- ----------|----------|----------|----------| --><p>All the motifs across the clusters were planted with a mutation rate of 0.</p>
</div>
<div id="input-and-feature-representation" class="section level2">
<h2 class="hasAnchor">
<a href="#input-and-feature-representation" class="anchor"></a>Input and feature representation</h2>
<p>We use one-hot encoding to represent the dinucleotide profiles of each sequence in the data set. <em>seqArchR</em> provides functions to read input from (a) a FASTA file, and (b) <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html">Biostrings::DNAStringSet</a></code> object.</p>
<div id="reading-input-as-fasta-file" class="section level3">
<h3 class="hasAnchor">
<a href="#reading-input-as-fasta-file" class="anchor"></a>Reading input as FASTA file</h3>
<p>The function <code><a href="../reference/prepare_data_from_FASTA.html">seqArchR::prepare_data_from_FASTA()</a></code> enables one-hot-encoding the DNA sequences in the given FASTA file. The one-hot-encoded sequences are returned as a sparse matrix with as many columns as the number of sequences in the FASTA file and (sequence length x <span class="math inline">\(4^{2}\)</span>) rows when dinucleotide profiles is selected. The number of rows will be (sequence length x <span class="math inline">\(4\)</span>) when mononucleotide profiles is selected. See the <code>sinuc_or_dinuc</code> argument.</p>
<p>Upon setting the logical argument <code>rawSeq</code> to <code>TRUE</code>, the function returns the raw sequences as a <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html">Biostrings::DNAStringSet</a></code> object, with <code>FALSE</code> it returns the column-wise one-hot encoded representation as noted above. When <code>raw_seq</code> is <code>TRUE</code>, <code>sinuc_or_dinuc</code> argument is ignored.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Creation of one-hot encoded data matrix from FASTA file</span>
<span class="va">inputFname</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"example_data.fa"</span>, 
                                  package <span class="op">=</span> <span class="st">"seqArchR"</span>, 
                                  mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># Specifying `dinuc` generates dinucleotide features</span>
<span class="va">inputSeqsMat</span> <span class="op">&lt;-</span> <span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="../reference/prepare_data_from_FASTA.html">prepare_data_from_FASTA</a></span><span class="op">(</span>fasta_fname <span class="op">=</span> <span class="va">inputFname</span>,
                                                  sinuc_or_dinuc <span class="op">=</span> <span class="st">"dinuc"</span><span class="op">)</span>
<span class="co">#&gt; Sequences OK,</span>
<span class="co">#&gt; Read 200 sequences</span>
<span class="co">#&gt; Generating dinucleotide profiles</span>

<span class="va">inputSeqsRaw</span> <span class="op">&lt;-</span> <span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="../reference/prepare_data_from_FASTA.html">prepare_data_from_FASTA</a></span><span class="op">(</span>fasta_fname <span class="op">=</span> <span class="va">inputFname</span>, 
                                               raw_seq <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="va">nSeqs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">inputSeqsRaw</span><span class="op">)</span>
<span class="va">positions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu">width</span><span class="op">(</span><span class="va">inputSeqsRaw</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="reading-input-as-a-dnastringset-object" class="section level3">
<h3 class="hasAnchor">
<a href="#reading-input-as-a-dnastringset-object" class="anchor"></a>Reading input as a DNAStringSet object</h3>
<p>If you already have a <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html">Biostrings::DNAStringSet</a></code> object, you can use the <code><a href="../reference/get_one_hot_encoded_seqs.html">get_one_hot_encoded_seqs()</a></code> function which directly accepts a <code>DNAStringSet</code> object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Creation of one-hot encoded data matrix from a DNAStringSet object</span>
<span class="va">inputSeqs_direct</span> <span class="op">&lt;-</span> <span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="../reference/get_one_hot_encoded_seqs.html">get_one_hot_encoded_seqs</a></span><span class="op">(</span>seqs <span class="op">=</span> <span class="va">inputSeqsRaw</span>, 
                                                  sinuc_or_dinuc <span class="op">=</span> <span class="st">"dinuc"</span><span class="op">)</span>
<span class="co">#&gt; Generating dinucleotide profiles</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span><span class="op">(</span><span class="va">inputSeqs_direct</span>, <span class="va">inputSeqsMat</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
</div>
</div>
<div id="visualize-input-sequences-as-an-image" class="section level2">
<h2 class="hasAnchor">
<a href="#visualize-input-sequences-as-an-image" class="anchor"></a>Visualize input sequences as an image</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Visualize the sequences in a image matrix where the DNA bases are </span>
<span class="co"># assigned fixed colors</span>

<span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="../reference/viz_seqs_acgt_mat.html">viz_seqs_acgt_mat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">inputSeqsRaw</span><span class="op">)</span>, 
                          pos_lab <span class="op">=</span> <span class="va">positions</span>, save_fname <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<p><img src="seqArchR_files/figure-html/plot-seqs-1.png" width="384"></p>
</div>
<div id="calling-seqarchr" class="section level2">
<h2 class="hasAnchor">
<a href="#calling-seqarchr" class="anchor"></a>Calling <em>seqArchR</em>
</h2>
<p>Setup seqArchR configuration as follows.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Set seqArchR configuration</span>
<span class="va">seqArchRconfig</span> <span class="op">&lt;-</span> <span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="../reference/set_config.html">set_config</a></span><span class="op">(</span>
        parallelize <span class="op">=</span> <span class="cn">TRUE</span>,
        n_cores <span class="op">=</span> <span class="fl">2</span>,
        n_runs <span class="op">=</span> <span class="fl">100</span>,
        k_min <span class="op">=</span> <span class="fl">1</span>,
        k_max <span class="op">=</span> <span class="fl">20</span>,
        mod_sel_type <span class="op">=</span> <span class="st">"stability"</span>,
        bound <span class="op">=</span> <span class="fl">10</span><span class="op">^</span><span class="op">-</span><span class="fl">6</span>,
        chunk_size <span class="op">=</span> <span class="fl">100</span>,
        result_aggl <span class="op">=</span> <span class="st">"ward.D"</span>, 
        result_dist <span class="op">=</span> <span class="st">"euclid"</span>,
        flags <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>debug <span class="op">=</span> <span class="cn">FALSE</span>, time <span class="op">=</span> <span class="cn">TRUE</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span>,
                     plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p>Once the configuration is setup, call the <code><a href="../reference/seqArchR.html">seqArchR::seqArchR</a></code> function with user-specified iterations.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Call/Run seqArchR</span>
<span class="va">seqArchRresult</span> <span class="op">&lt;-</span> <span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="../reference/seqArchR.html">seqArchR</a></span><span class="op">(</span>config <span class="op">=</span> <span class="va">seqArchRconfig</span>,
                            seqs_ohe_mat <span class="op">=</span> <span class="va">inputSeqsMat</span>,
                            seqs_raw <span class="op">=</span> <span class="va">inputSeqsRaw</span>,
                            seqs_pos <span class="op">=</span> <span class="va">positions</span>,
                            total_itr <span class="op">=</span> <span class="fl">2</span>,
                            set_ocollation <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; ── Setting up ──────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt; ℹ Parallelization: 2 cores</span>
<span class="co">#&gt; ℹ Model selection by factor stability</span>
<span class="co">#&gt; ℹ Bound: 1e-06</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Iteration 1 of 2 [1 chunk] ──────────────────────────────────────────────────</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Outer chunk 1 of 1 [Size: 200] ──</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Inner chunk 1 of 2 [Size: 100]</span>
<span class="co">#&gt; Checking K =2</span>
<span class="co">#&gt; Checking K =3</span>
<span class="co">#&gt; Checking K =4</span>
<span class="co">#&gt; Checking K =5</span>
<span class="co">#&gt; ℹ Best K for this chunk: 4</span>
<span class="co">#&gt; ℹ Adjusting for overfitting, fetched 3 clusters</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Inner chunk 2 of 2 [Size: 100]</span>
<span class="co">#&gt; Checking K =2</span>
<span class="co">#&gt; Checking K =3</span>
<span class="co">#&gt; Checking K =4</span>
<span class="co">#&gt; Checking K =5</span>
<span class="co">#&gt; ℹ Best K for this chunk: 4</span>
<span class="co">#&gt; ✔ 1 of 1 outer chunk complete</span>
<span class="co">#&gt; ✔ 1 of 2 iterations complete</span>
<span class="co">#&gt; → Iteration 1 completed: 1m 9.7s</span>
<span class="co">#&gt; → Time ellapsed since start: 1m 9.8s</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Iteration 2 of 2 [4 chunks] ─────────────────────────────────────────────────</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Outer chunk 1 of 4 [Size: 50] ──</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Inner chunk 1 of 1 [Size: 50]</span>
<span class="co">#&gt; Checking K =2</span>
<span class="co">#&gt; ℹ Best K for this chunk: 1</span>
<span class="co">#&gt; ✔ 1 of 4 outer chunks complete</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Outer chunk 2 of 4 [Size: 70] ──</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Inner chunk 1 of 1 [Size: 70]</span>
<span class="co">#&gt; Checking K =2</span>
<span class="co">#&gt; Checking K =3</span>
<span class="co">#&gt; ℹ Best K for this chunk: 2</span>
<span class="co">#&gt; ✔ 2 of 4 outer chunks complete</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Outer chunk 3 of 4 [Size: 29] ──</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Inner chunk 1 of 1 [Size: 29]</span>
<span class="co">#&gt; Checking K =2</span>
<span class="co">#&gt; ℹ Best K for this chunk: 1</span>
<span class="co">#&gt; ✔ 3 of 4 outer chunks complete</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Outer chunk 4 of 4 [Size: 51] ──</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Inner chunk 1 of 1 [Size: 51]</span>
<span class="co">#&gt; Checking K =2</span>
<span class="co">#&gt; ℹ Best K for this chunk: 1</span>
<span class="co">#&gt; ✔ 4 of 4 outer chunks complete</span>
<span class="co">#&gt; ✔ 2 of 2 iterations complete</span>
<span class="co">#&gt; → Iteration 2 completed: 1m 8s</span>
<span class="co">#&gt; → Time ellapsed since start: 2m 17.7s</span>
<span class="co">#&gt; ── seqArchR exiting</span>
<span class="co">#&gt; 2m 17.8s ───────────────────────────────────────────────────</span></code></pre></div>
</div>
<div id="understanding-the-result-object-from-seqarchr" class="section level2">
<h2 class="hasAnchor">
<a href="#understanding-the-result-object-from-seqarchr" class="anchor"></a>Understanding the result object from <em>seqArchR</em>
</h2>
<p>In the version 0.99.0, <em>seqArchR</em> naively returns a result object which is a nested list of seven elements. These include: - the sequence cluster labels per iteration [<code>seqsClustLabels</code>]; - the collection of NMF basis vectors per iteration [<code>clustBasisVectors</code>]: each is a list of two elements <code>nBasisVectors</code> and <code>basisVectors</code>; - the clustering solution, [<code>clustSol</code>], which is obtained upon combining raw clusters from the last iteration of seqArchR. This element stores the clustering of NMF basis vectors [<code>basisVectorsClust</code>] and the sequence clusters [<code>clusters</code>]; - the raw sequences provided [<code>rawSeqs</code>]; - if timeFlag is set, timing information (in minutes) per iteration [<code>timeInfo</code>]; - the configuration setting [<code>config</code>]; and - the call itself [<code>call</code>].</p>
<div id="nmf-basis-vectors" class="section level3">
<h3 class="hasAnchor">
<a href="#nmf-basis-vectors" class="anchor"></a>NMF basis vectors</h3>
<p><em>seqArchR</em> stores the NMF basis vectors corresponding to each cluster in every iteration in the variable <code>clustBasisVectors</code>. <code>clustBasisVectors</code> is a numbered list corresponding to the number of iterations performed. This is then again a list holding two pieces of information: the number of basis vectors (<code>nBasisVectors</code>) and the basis vectors (<code>basisVectors</code>).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># Basis vectors at iteration 2</span>
<span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="../reference/get_clBasVec.html">get_clBasVec_k</a></span><span class="op">(</span><span class="va">seqArchRresult</span>, iter<span class="op">=</span><span class="fl">2</span><span class="op">)</span>
<span class="co">#&gt; [1] 5</span>

<span class="va">i2_bv</span> <span class="op">&lt;-</span> <span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="../reference/get_clBasVec.html">get_clBasVec_m</a></span><span class="op">(</span><span class="va">seqArchRresult</span>, iter<span class="op">=</span><span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">i2_bv</span><span class="op">)</span>
<span class="co">#&gt; [1] 1600    5</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="va">i2_bv</span><span class="op">)</span>
<span class="co">#&gt;            [,1]       [,2]       [,3]       [,4]       [,5]</span>
<span class="co">#&gt; [1,] 0.05168607 0.07646511 0.05101696 0.08471968 0.12920686</span>
<span class="co">#&gt; [2,] 0.02486037 0.07603933 0.09368281 0.04243838 0.10288363</span>
<span class="co">#&gt; [3,] 0.05076204 0.04942229 0.04745960 0.04243838 0.07754562</span>
<span class="co">#&gt; [4,] 0.09945618 0.04225726 0.11945756 0.03847347 0.05154964</span>
<span class="co">#&gt; [5,] 0.07484865 0.09133385 0.17064509 0.04028088 0.09605441</span>
<span class="co">#&gt; [6,] 0.05051104 0.09435299 0.14381388 0.04232253 0.02514596</span></code></pre></div>
<p>The NMF basis vectors can be visualized as a heatmap and/or sequence logo using [<a href="https://snikumbh.github.io/seqArchR/reference/viz_bas_vec_heatmap_seqlogo.html" class="uri">https://snikumbh.github.io/seqArchR/reference/viz_bas_vec_heatmap_seqlogo.html</a>] (viz_bas_vec_heat_seqlogo) function.</p>
</div>
<div id="basis-vectors-at-iteration-1" class="section level3">
<h3 class="hasAnchor">
<a href="#basis-vectors-at-iteration-1" class="anchor"></a>Basis vectors at iteration 1</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="../reference/viz_bas_vec_heatmap_seqlogo.html">viz_bas_vec_heatmap_seqlogo</a></span><span class="op">(</span>feat_mat <span class="op">=</span> <span class="fu"><a href="../reference/get_clBasVec.html">get_clBasVec_m</a></span><span class="op">(</span><span class="va">seqArchRresult</span>, <span class="fl">1</span><span class="op">)</span>, 
                                  method <span class="op">=</span> <span class="st">"bits"</span>, sinuc_or_dinuc <span class="op">=</span> <span class="st">"dinuc"</span><span class="op">)</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span>
<span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">#&gt; replace the existing scale.</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span>
<span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">#&gt; replace the existing scale.</span></code></pre></div>
<p><img src="seqArchR_files/figure-html/viz-BV-1-1.png" width="2400"></p>
<pre><code>#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =
#&gt; "none")` instead.
#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will
#&gt; replace the existing scale.</code></pre>
<p><img src="seqArchR_files/figure-html/viz-BV-1-2.png" width="2400"></p>
<pre><code>#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =
#&gt; "none")` instead.
#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will
#&gt; replace the existing scale.</code></pre>
<p><img src="seqArchR_files/figure-html/viz-BV-1-3.png" width="2400"><img src="seqArchR_files/figure-html/viz-BV-1-4.png" width="2400"></p>
</div>
<div id="basis-vectors-at-iteration-2" class="section level3">
<h3 class="hasAnchor">
<a href="#basis-vectors-at-iteration-2" class="anchor"></a>Basis vectors at iteration 2</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="../reference/viz_bas_vec_heatmap_seqlogo.html">viz_bas_vec_heatmap_seqlogo</a></span><span class="op">(</span>feat_mat <span class="op">=</span> <span class="fu"><a href="../reference/get_clBasVec.html">get_clBasVec_m</a></span><span class="op">(</span><span class="va">seqArchRresult</span>, <span class="fl">2</span><span class="op">)</span>, 
                                  method <span class="op">=</span> <span class="st">"bits"</span>, sinuc_or_dinuc <span class="op">=</span> <span class="st">"dinuc"</span><span class="op">)</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span>
<span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">#&gt; replace the existing scale.</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span>
<span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">#&gt; replace the existing scale.</span></code></pre></div>
<p><img src="seqArchR_files/figure-html/viz-BV-2-1.png" width="2400"></p>
<pre><code>#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =
#&gt; "none")` instead.
#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will
#&gt; replace the existing scale.</code></pre>
<p><img src="seqArchR_files/figure-html/viz-BV-2-2.png" width="2400"></p>
<pre><code>#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =
#&gt; "none")` instead.
#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will
#&gt; replace the existing scale.</code></pre>
<p><img src="seqArchR_files/figure-html/viz-BV-2-3.png" width="2400"></p>
<pre><code>#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =
#&gt; "none")` instead.
#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will
#&gt; replace the existing scale.</code></pre>
<p><img src="seqArchR_files/figure-html/viz-BV-2-4.png" width="2400"><img src="seqArchR_files/figure-html/viz-BV-2-5.png" width="2400"></p>
</div>
<div id="visualize-sequences-by-clusters" class="section level3">
<h3 class="hasAnchor">
<a href="#visualize-sequences-by-clusters" class="anchor"></a>Visualize sequences by clusters</h3>
<p>The clustered output from <em>seqArchR</em> can again be visualized as a matrix. Use the <a href="seqs_str">https://snikumbh.github.io/seqArchR/reference/seqs_str.html</a> function to fetch sequences by clusters at any iteration and call <code>seqArchR::viz_seqs_as_acgt_mat</code> as shown.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="../reference/viz_seqs_acgt_mat.html">viz_seqs_acgt_mat</a></span><span class="op">(</span><span class="fu"><a href="../reference/seqs_str.html">seqs_str</a></span><span class="op">(</span><span class="va">seqArchRresult</span>, iter <span class="op">=</span> <span class="fl">1</span>, ord <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
                                  pos_lab <span class="op">=</span> <span class="va">positions</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="seqArchR_files/figure-html/clust-itr1-1.png" alt="Figure: Clusters at iteration 1" width="384"><p class="caption">
Figure: Clusters at iteration 1
</p>
</div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu">seqArchR</span><span class="fu">::</span><span class="fu"><a href="../reference/viz_seqs_acgt_mat.html">viz_seqs_acgt_mat</a></span><span class="op">(</span><span class="fu"><a href="../reference/seqs_str.html">seqs_str</a></span><span class="op">(</span><span class="va">seqArchRresult</span>, iter <span class="op">=</span> <span class="fl">2</span>, ord <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
                                  pos_lab <span class="op">=</span> <span class="va">positions</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="seqArchR_files/figure-html/clust-itr2-1.png" alt="Figure: Clusters at iteration 2" width="384"><p class="caption">
Figure: Clusters at iteration 2
</p>
</div>
</div>
</div>
</div>
<div id="conclusion" class="section level1">
<h1 class="hasAnchor">
<a href="#conclusion" class="anchor"></a>Conclusion</h1>
<p><em>seqArchR</em> can detect <em>de novo</em> sequence features and simultaneously identify the complex interactions of different features together with their positional specificities.</p>
<p>Note that the sequence architectures identified by <em>seqArchR</em> have no limitations due to the size of the motifs or gaps in them, distance between motifs, compositional and positional variations in the individual motifs and their effects on the complex interactions, and number of motifs involved in any interaction.</p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] Biostrings_2.62.0   GenomeInfoDb_1.30.0 XVector_0.34.0     </span>
<span class="co">#&gt; [4] IRanges_2.28.0      S4Vectors_0.32.3    BiocGenerics_0.40.0</span>
<span class="co">#&gt; [7] seqArchR_0.99.0    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_1.0.7             lattice_0.20-45        prettyunits_1.1.1     </span>
<span class="co">#&gt;  [4] ggseqlogo_0.1          rprojroot_2.0.2        digest_0.6.29         </span>
<span class="co">#&gt;  [7] utf8_1.2.2             R6_2.5.1               plyr_1.8.6            </span>
<span class="co">#&gt; [10] evaluate_0.14          ggplot2_3.3.5          highr_0.9             </span>
<span class="co">#&gt; [13] pillar_1.6.4           zlibbioc_1.40.0        rlang_0.4.12          </span>
<span class="co">#&gt; [16] Matrix_1.3-4           rmarkdown_2.11         pkgdown_1.6.1         </span>
<span class="co">#&gt; [19] textshaping_0.3.6      desc_1.4.0             labeling_0.4.2        </span>
<span class="co">#&gt; [22] stringr_1.4.0          RCurl_1.98-1.5         munsell_0.5.0         </span>
<span class="co">#&gt; [25] compiler_4.1.2         xfun_0.28              pkgconfig_2.0.3       </span>
<span class="co">#&gt; [28] systemfonts_1.0.2      htmltools_0.5.2        tibble_3.1.6          </span>
<span class="co">#&gt; [31] GenomeInfoDbData_1.2.7 matrixStats_0.61.0     fansi_0.5.0           </span>
<span class="co">#&gt; [34] crayon_1.4.2           bitops_1.0-7           grid_4.1.2            </span>
<span class="co">#&gt; [37] gtable_0.3.0           lifecycle_1.0.1        magrittr_2.0.1        </span>
<span class="co">#&gt; [40] scales_1.1.1           cli_3.1.0              stringi_1.7.6         </span>
<span class="co">#&gt; [43] cachem_1.0.6           farver_2.1.0           reshape2_1.4.4        </span>
<span class="co">#&gt; [46] fs_1.5.0               ellipsis_0.3.2         ragg_1.1.3            </span>
<span class="co">#&gt; [49] vctrs_0.3.8            cowplot_1.1.1          tools_4.1.2           </span>
<span class="co">#&gt; [52] glue_1.5.1             parallel_4.1.2         fastmap_1.1.0         </span>
<span class="co">#&gt; [55] yaml_2.2.1             colorspace_2.0-2       cluster_2.1.2         </span>
<span class="co">#&gt; [58] memoise_2.0.1          knitr_1.36</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sarvesh Nikumbh.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
